AWSAppSyncClient/AWSAppSyncClient.swift

```diff
@@ -66,8 +66,8 @@ public class AWSAppSyncClient {
         try self.init(appSyncConfig: appSyncConfig, reachabilityFactory: nil)
     }
 
-    init(appSyncConfig: AWSAppSyncClientConfiguration,
-         reachabilityFactory: NetworkReachabilityProvidingFactory.Type? = nil) throws {
+    public init(appSyncConfig: AWSAppSyncClientConfiguration,
```

Comment:
I agree in principle with the change, but in order to make the intent clear, let's add some documentation along the lines of:

```swift
/// Creates an AWSAppSyncClient with the specified configuration and NetworkReachabilityProvidingFactory.
/// This method is primarily intended to facilitate integration testing, but may be used by apps that wish
/// to use their own reachability solution in place of AppSync's bundled reachability framework.
///
/// Note that the AppSync client's use of the Reachability client provided by the factory should be
/// considered an implementation detail. In particular, apps should not rely on AppSync to inspect network
/// reachability status before attempting a network connection.
///
/// - Parameters:
///   - appSyncConfig: The `AWSAppSyncClientConfiguration` object.
///   - reachabilityFactory: The factory that provides `NetworkReachabilityProviding` instances
```